pages/index.tsx

```diff
@@ -31,15 +33,21 @@ export default function Home() {
 
   useEffect(() => {
     console.log('effect');
-    const parsedGiftData = JSON.parse(getLocalStorage('giftData'));
-    const gifts = sortGiftsOldestFirst(parsedGiftData);
-    setGiftData(gifts);
+    async function fetchData() {
+      const gifts = await (await jsonServerFunctions.getAll()).data;
+      setGiftData(gifts);
+    }
+    fetchData().catch((e) => {
+      throw new Error(e);
```

Comment:
Tässä ei voi heittää virhettä eteenpäin, koska mikään ei nappaa sitä enää kiinni ja käsittele. Parempi tulostaa virhe konsoliin. Halutessaan voisi myös näyttää käyttäjälle virheilmoituksen esim. toastin avulla, mutta se kannattaa tehdä toisessa PR:ssä.
```suggestion
      console.error(e);
```